src/lang/KclSingleton.ts

```diff
@@ -300,24 +300,30 @@ export class KclManager {
     this._kclErrors = errors
     this._programMemory = programMemory
     if (updates !== 'artifactRanges') return
-    Object.entries(this.engineCommandManager.artifactMap).forEach(
+
+    // TODO the below seems like a work around, I wish there's a comment explaining exactly what
+    // problem this solves, but either way we should strive to remove it.
+    Array.from(this.engineCommandManager.artifactGraph).forEach(
```

Comment:
You can iterate over maps directly using `for...of`. You don't need `Array.from()`. I guess it's arbitrary in this case, and maybe `forEach()` is preferred to minimize the diff. But I generally prefer `for...of` over `forEach()` because you can `break`, return from the enclosing function, etc.